runtime/src/tests.rs

```diff
@@ -130,6 +138,156 @@ fn new_test_ext() -> sp_io::TestExternalities {
+        let multi = Multisig::multi_account_id(&account_ids[..3], 2);
+
+        let call = call_transfer(account_ids[2].clone(), 500).encode();
+        let hash = blake2_256(&call);
+
+        assert_ok!(Balances::transfer_allow_death(RuntimeOrigin::signed(account_ids[0].clone()), MultiAddress::Id(multi.clone()), 500));
+        assert_ok!(Balances::transfer_allow_death(RuntimeOrigin::signed(account_ids[1].clone()), MultiAddress::Id(multi.clone()), 500));
+        assert_ok!(Balances::transfer_allow_death(RuntimeOrigin::signed(account_ids[2].clone()), MultiAddress::Id(multi.clone()), 500));
+
+        assert_ok!(Multisig::approve_as_multi(
+			RuntimeOrigin::signed(account_ids[0].clone()),
+			3,
+			vec![account_ids[1].clone(), account_ids[2].clone()],
+			None,
+			hash,
+			Weight::zero()
+		));
+
+        assert_ok!(Multisig::approve_as_multi(
+			RuntimeOrigin::signed(account_ids[1].clone()),
+			3,
+			vec![account_ids[0].clone(), account_ids[2].clone()],
+			Some(Multisig::timepoint()),
+			hash,
+			Weight::zero()
+		));
+
+        assert_eq!(Balances::free_balance(multi.clone()), 1500);
+        assert_eq!(Balances::free_balance(account_ids[0].clone()), testsfixtures::SAMPLE_USERS[0].starting_balance - 1000);
+        assert_eq!(Balances::reserved_balance(account_ids[0].clone()), 500);
+
+        assert_ok!(Multisig::cancel_as_multi(
+            RuntimeOrigin::signed(account_ids[0].clone()),
+            3,
+            vec![account_ids[1].clone(), account_ids[2].clone()],
+            Multisig::timepoint(),
+            hash
+        ));
+
+        assert_eq!(Balances::free_balance(multi.clone()), 1500);
+        assert_eq!(Balances::free_balance(account_ids[0].clone()), testsfixtures::SAMPLE_USERS[0].starting_balance - 500);
+        assert_eq!(Balances::reserved_balance(account_ids[0].clone()), 0);
+    });
+}
+
+#[test]
+fn test_set_sudo_to_multisig() {
+    new_test_ext().execute_with(|| {
+        // Extract the account IDs from SAMPLE_USERS
+        let account_ids: Vec<AccountId32> = testsfixtures::SAMPLE_USERS.iter().map(|user| AccountId32::from(user.raw_account)).collect();
+        let multi = Multisig::multi_account_id(&account_ids[..3], 2);
+
+        assert_ok!(Balances::transfer_allow_death(RuntimeOrigin::signed(account_ids[0].clone()), MultiAddress::Id(multi.clone()), 1000));
+
+        // Check existing sudo key
+        let existing_sudo_key = Sudo::key();
+        assert_eq!(existing_sudo_key, Some(account_ids[0].clone()));
+
+        // Setting the multisig account as the new sudo account
+        assert_ok!(Sudo::set_key(RuntimeOrigin::signed(account_ids[0].clone()), MultiAddress::Id(multi.clone())));
+
+        // Ensure the sudo key is set correctly
+        assert_eq!(Sudo::key(), Some(multi.clone()));
+
+        // Prepare a sudo call to change the sudo key again to a new account (account_ids[1])
+        let sudo_call = RuntimeCall::Sudo(SudoCall::set_key {
```

Comment:
use `pallet_sudo::Call` and remove the unneeded indiretion here (you use it just once)